src/live.rs

```diff
@@ -647,6 +648,53 @@ fn extract_cpio(buf: &[u8]) -> Result<Vec<u8>> {
     }
 }
 
+#[derive(Serialize)]
+struct IsoInspectOutput {
+    header: IsoFs,
+    records: Vec<String>,
+}
+
+pub fn iso_inspect(config: &IsoInspectConfig) -> Result<()> {
+    let mut iso = IsoFs::new(&config.input)?;
+    let mut records: Vec<String> = Vec::new();
+    let root = iso.get_root_directory()?;
+    collect_iso_records(&mut iso, "", &root, &mut records).context("collecting ISO records")?;
+
+    let inspect_out = IsoInspectOutput {
+        header: iso,
+        records,
+    };
+
+    let stdout = io::stdout();
+    let mut out = stdout.lock();
+    serde_json::to_writer_pretty(&mut out, &inspect_out)
+        .context("failed to serialize ISO metadata")?;
+    out.write_all(b"\n").context("failed to write newline")?;
+    Ok(())
+}
+
+fn collect_iso_records(
```

Comment:
I may fold this into the `iso9660` module in a future patch.

Reply:
Did this now as a new iterator accessible via `IsoFs::walk()`. The function is a bit gnarlier than I would've liked, but the interface for callers is much nicer.